tests/unit/Command/FixEncryptedVersionTest.php

```diff
@@ -0,0 +1,201 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Encryption\Tests\Command;
+
+use OC\Encryption\Util;
+use OC\Files\Cache\Cache;
+use OC\Files\View;
+use OCA\Encryption\Command\FixEncryptedVersion;
+use OCP\Files\IRootFolder;
+use OCP\IUserManager;
+use Symfony\Component\Console\Tester\CommandTester;
+use Test\TestCase;
+
+/**
+ * Class FixEncryptedVersionTest
+ *
+ * @group DB
+ * @package OCA\Encryption\Tests\Command
+ */
+class FixEncryptedVersionTest extends TestCase {
+	const TEST_ENCRYPTION_VERSION_AFFECTED_USER = 'test_enc_version_affected_user1';
+
+	/** @var IRootFolder */
+	private $rootFolder;
+
+	/** @var IUserManager */
+	private $userManager;
+
+	/** @var View */
+	private $view;
+
+	/** @var FixEncryptedVersion */
+	private $fixEncryptedVersion;
+
+	/** @var CommandTester */
+	private $commandTester;
+
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+		\OC::$server->getUserManager()->createUser(self::TEST_ENCRYPTION_VERSION_AFFECTED_USER, 'foo');
+		//Enable encryption app
+		\OC_App::enable("encryption");
+		//Enable encryption
+		\OC::$server->getConfig()->setAppValue('core', 'encryption_enabled', 'yes');
+		//Enable Masterkey
+		\OC::$server->getConfig()->setAppValue('encryption', 'useMasterKey', '1');
+		\OC::$server->getUserSession()->logout();
+	}
+
+	public static function tearDownAfterClass() {
+		parent::tearDownAfterClass();
+		$user = \OC::$server->getUserManager()->get(self::TEST_ENCRYPTION_VERSION_AFFECTED_USER);
+		if ($user !== null) {
+			$user->delete();
+		}
+		\OC_App::disable('encryption');
+		\OC::$server->getConfig()->deleteAppValue('core', 'encryption_enabled');
+		\OC::$server->getConfig()->deleteAppValue('core', 'default_encryption_module');
+		\OC::$server->getConfig()->deleteAppValues('encryption');
+	}
+
+	public function setUp() {
+		parent::setUp();
+		$this->rootFolder = \OC::$server->getRootFolder();
+		$this->userManager = \OC::$server->getUserManager();
+		$this->view = new View("");
+		$this->fixEncryptedVersion = new FixEncryptedVersion($this->rootFolder, $this->userManager, $this->view);
+		$this->commandTester = new CommandTester($this->fixEncryptedVersion);
+	}
+
+	/**
+	 * In this test the encrypted version is set to zero whereas it should have been
+	 * set to a positive non zero number.
+	 */
+	public function testEncryptedVersionZero() {
+		$this->markTestSkipped("Skipping this test because the user login would fail in the CI. And once we have a db endpoints available in acceptance test, we would add the test over there");
```

Comment:
please continue researching a bit

you should use `$this->loginAsUser()` for a more accurate login simulation. see https://github.com/owncloud/core/blob/v10.1.1/tests/lib/TestCase.php#L368